modules/indexer/issues/elastic_search.go

```diff
@@ -20,8 +21,11 @@ var _ Indexer = &ElasticSearchIndexer{}
 
 // ElasticSearchIndexer implements Indexer interface
 type ElasticSearchIndexer struct {
-	client      *elastic.Client
-	indexerName string
+	client               *elastic.Client
+	indexerName          string
+	available            bool
+	availabilityCallback func(bool)
```

Comment:
Unfortunately this needs to be locked unless it can be set in NewElasticSearchIndexer, or before the availability checker go routine starts and before checkError or other calls to checkAvailability.

Reply:
SetAvailabilityChangeCallback is called single time during queue initialization so there should not be race for it but I can add single lock for both